### PR TITLE
TensorFlow-Free Data Loader

### PR DESCRIPTION
**Pull request by jpvantassel**
# In this PR

- ✨ Create tensorflow-free dataloader. Data loader can handle any number of training simulations of arbitrary length.

### CODE REVIEW

**Review by kks32 — COMMENTED** [review]
If we are not calling distribute.py in this version, could we not include it and wait until we have it completed? Then, we can merge?

**Comment by jpvantassel** [comment]
> If we are not calling distribute.py in this version, could we not include it and wait until we have it completed? Then, we can merge?

`distribute.py` has been removed.

**Comment by jpvantassel** [comment]
> This is wonderful! Thanks @jpvantassel
> 
>     * Could you add some info in the README about the format of `.npz` files?
> 
>     * I don't see a convertor you mentioned for TFRecord to .npz

Notes on the PyTorch dataloader and npz format has been added to README.

Mention was in relation to the conversion of tfrecord to npz; there is no converter.
The branch tfrecord_to_npy includes details on the transformation.

File: gns/data_loader.py
Comment (kks32) [inline gns/data_loader.py:42]: This will only work for 2D? Can we keep this as an input argument?

Also, the number of particles?

**Review by kks32 — CHANGES_REQUESTED** [review]
Other than the question about hardcoding the particle dimension, I'm good.

**Comment by jpvantassel** [comment]
> Could we generalize the data loader for `nparticles` and `dim`?

Done, see c406410

**Comment by jpvantassel** [comment]
> Other than the question about hardcoding the particle dimension, I'm good.

**Review by kks32 — APPROVED** [review]

**Pull request merged**
